apps/frontend/src/libs/modules/store/libs/middlewares/handle-redirect.middleware.ts

```diff
@@ -0,0 +1,37 @@
+import { isAction, type Middleware } from "@reduxjs/toolkit";
+
+import { AppRoute } from "~/libs/enums/enums.js";
+import { store } from "~/libs/modules/store/store.js";
+import { actions as appActions } from "~/modules/app/app.js";
+import { actions as authActions } from "~/modules/auth/auth.js";
+
+type Properties = {
+	action: unknown;
+	next: (action: unknown) => unknown;
+};
+
+const handleErrorMiddleware = ({ action, next }: Properties): unknown => {
+	if (
+		isAction(action) &&
+		(action.type === authActions.resetPassword.fulfilled.type ||
+			action.type === authActions.checkIsResetPasswordExpired.rejected.type)
+	) {
```

Comment:
You don't need this middleware. For `fulfilled` state, do the redirection in action. For `rejected` state in handleErrorMiddleware